client/lobby/OptionsTab.cpp

```diff
@@ -609,21 +612,24 @@ void OptionsTab::SelectionWindow::genContentHeroes()
 		{
 
 			int x = i % elementsPerLine;
-			int y = i / elementsPerLine;
+			int y = (i / elementsPerLine) - sliderLine;
 
 			PlayerSettings set = PlayerSettings();
 			set.hero = elem;
 
 			CPlayerSettingsHelper helper = CPlayerSettingsHelper(set, SelType::HERO);
 
-			components.push_back(std::make_shared<CAnimImage>(helper.getImageName(true), helper.getImageIndex(true), 0, x * (ICON_BIG_WIDTH-1), y * (ICON_BIG_HEIGHT-1)));
-			drawOutlinedText(x * (ICON_BIG_WIDTH-1) + TEXT_POS_X, y * (ICON_BIG_HEIGHT-1) + TEXT_POS_Y, (selectedHero == elem) ? Colors::YELLOW : Colors::WHITE, helper.getName());
-			ImagePath image = ImagePath::builtin("lobby/townBorderBig");
-			if(selectedHero == elem)
-				image = ImagePath::builtin("lobby/townBorderBigActivated");
-			if(unusableHeroes.count(elem))
-				image = ImagePath::builtin("lobby/townBorderBigGrayedOut");
-			components.push_back(std::make_shared<CPicture>(image, x * (ICON_BIG_WIDTH-1), y * (ICON_BIG_HEIGHT-1)));
+			if(y >= 0 && y <= MAX_LINES - 1)
```

Comment:
Perhaps use early return (or `continue` since this is a loop) instead of nesting multiple if's?

Same for pre-existing `type->heroClass->faction == selectedFaction` check above